src/stim/circuit/gate_data.h

```diff
@@ -136,12 +328,33 @@ struct ExtraGateData {
         const char *h_s_cx_m_r_decomposition);
 };
 
+template <typename SIMULATOR>
+struct GateVTable {
+   private:
+    using sim_func_ptr_t = void (SIMULATOR::*)(const OperationData&);
+    std::array<sim_func_ptr_t, 256> funcs;
+    constexpr void add_simulator_function(Gates gate_id, sim_func_ptr_t simulator_function);
+    constexpr void add_simulator_function_annotations();
+    constexpr void add_simulator_function_blocks();
+    constexpr void add_simulator_function_collapsing();
+    constexpr void add_simulator_function_controlled();
```

Comment:
Is this method actually needed as part of the definition? I can see that implementations of the template would want to do this kind of grouping, but it seems odd to *require* it.